src/core/tsi/ssl_transport_security.cc

```diff
@@ -1049,25 +1049,22 @@ void tsi_ssl_session_cache_unref(tsi_ssl_session_cache* cache) {
   reinterpret_cast<tsi::SslSessionLRUCache*>(cache)->Unref();
 }
 
-/* --- tsi_frame_protector methods implementation. ---*/
-
-static tsi_result ssl_protector_protect(tsi_frame_protector* self,
-                                        const unsigned char* unprotected_bytes,
-                                        size_t* unprotected_bytes_size,
-                                        unsigned char* protected_output_frames,
-                                        size_t* protected_output_frames_size) {
-  tsi_ssl_frame_protector* impl =
-      reinterpret_cast<tsi_ssl_frame_protector*>(self);
+/* --- tsi_frame_protector util methods implementation. ---*/
+tsi_result ssl_protector_protect_util(
```

Comment:
See https://github.com/grpc/grpc/blob/master/include/grpcpp/security/tls_certificate_verifier.h#L89-L102 for an example to for better API contract documentation

Reply:
Thanks for the review Zhen! These functions rely on internal functions in `ssl_transport_security.cc` (e.g., `ssl_write`). Moving them to a new file requires us to isolate all those helper functions into the same new file as well. I was trying to minimize the change here. What do you think of keeping them inside `ssl_transport_security.h/cc`?

Reply:
It seems `do_ssl_write` is also shared within two functions, so I think it makes sense to move them under the utility folder as well.
Ideally, I'd like to restructure everything so that we could use class-based techniques to arrange everything, but that's beyond the scope of this pull request. Would you mind moving them into a utility library first as an intermediate solution?